servicediscovery/mdns/mdns.go

```diff
@@ -0,0 +1,75 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package mdns
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/dapr/components-contrib/servicediscovery"
+	"github.com/grandcat/zeroconf"
+)
+
+func NewMDNSResolver() *Resolver {
+	return &Resolver{}
+}
+
+type Resolver struct {
+	Once     sync.Once
+	Resolver *zeroconf.Resolver
+}
+
+func (z *Resolver) ResolveID(req *servicediscovery.ResolveRequest) (string, error) {
+	port, err := z.LookupPortMDNS(req.ID)
+	if err != nil {
+		return "", err
+	}
+	return fmt.Sprintf("localhost:%v", port), nil
+}
+
+// LookupPortMDNS uses mdns to find the port of a given service entry on a local network
+func (z *Resolver) LookupPortMDNS(id string) (int, error) {
+	var err error
+	var t *zeroconf.Resolver
```

Comment:
we do not need to declare the variable here. instead we can use `:=` assigner inside once.do {}

Reply:
This needs to be outside for the same reason as the comment above.